nautobot/docs/development/core/kubernetes-job-support.md

```diff
@@ -0,0 +1,509 @@
+
+```bash
+NAME                                                STATUS     COMPLETIONS   DURATION   AGE
+nautobot-job-7442d540-10a5-4123-95f6-d7e2899012b2   Complete   1/1           8s         11s
+```
+
+In order to confirm the job is completed successfully from the perspective of nautobot, we can again use `nbshell` and examine the latest job result:
+
+```py
+# Django version 4.2.16
+# Nautobot version 2.4.0a1
+# Example Nautobot App version 1.0.0
+Python 3.11.10 (main, Nov 12 2024, 22:19:37) [GCC 12.2.0] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+(InteractiveConsole)
+>>> JobResult.objects.latest()
+<JobResult: Export Object List started at 2024-11-20 21:35:49.812765+00:00 (SUCCESS)>
+```
+
+The job is completed successfully as indicated by the job result status.
+
+### Running a Scheduled Job
+
+We can run scheduled jobs as well. In order to run scheduled jobs, we do need celery beat which should already be running from previous steps. To confirm that the celery beat is running, we need to retrieve the celery beat pod name by running the command `kubectl get pods`. You should see the following output or something similar, copy the pod name with prefix `celery-beat-*`.
+
+```bash
+NAME                           READY   STATUS    RESTARTS   AGE
+celery-beat-6fb67477b7-rsw62   1/1     Running   0          10m
+db-8687b48964-gtvtc            1/1     Running   0          10m
+nautobot-679bdc765-pl2ld       1/1     Running   0          10m
+redis-7cc58577c-tl5sq          1/1     Running   0          10m
+```
+
+we can run the command `kubectl logs <celery-beat-pod-name> -f` and you should see the following output or something similar:
+
+```bash
+LocalTime -> 2024-11-20 18:49:41
+Configuration ->
+    . broker -> redis://:**@redis:6379/0
+    . loader -> celery.loaders.app.AppLoader
+    . scheduler -> nautobot.core.celery.schedulers.NautobotDatabaseScheduler
+
+    . logfile -> [stderr]@%INFO
+    . maxinterval -> 5.00 seconds (5s)
+[2024-11-20 18:49:41,161: INFO/MainProcess] beat: Starting...
+```
+
+Open another terminal and execute the command `kubectl exec --stdin --tty <nautobot-pod-name> -- /bin/bash` and run `nautobot-server nbshell`. We are going to create a new Scheduled Export Object List Job.
+
+First we import all the required packages:
+
+```py
+>>> from nautobot.extras.choices import JobExecutionType
+>>> from datetime import datetime
+>>> from zoneinfo import ZoneInfo
+```
+
+Then we can create the Scheduled Job
+
+```py
+>>> ScheduledJob.objects.create(name="Export Object List Hourly", task="nautobot.core.jobs.ExportObjectList", job_model=Job.objects.ge
+t(name="Export Object List"), interval=JobExecutionType.TYPE_HOURLY, user=User.objects.first(), approval_required=False, start_time=da
+tetime.now(ZoneInfo("America/New_York")), time_zone=ZoneInfo("America/New_York"), job_queue=JobQueue.objects.get(name="kubernetes"), kwargs={"content_type": 1})
```

Comment:
Odd line wrapping here?

Reply:
Probably because it should be ready to copy & paste into own terminal. But yea, approach with creating an object and asserting values line by line would be more readable :) 
```
scheduled_job = ScheduledJob()
scheduled_job.name = "..."
```